lib/reconcile/utils.go

```diff
@@ -580,3 +661,55 @@ func getBaseAstarteAPIURL(cr *apiv1alpha1.Astarte) string {
 
 	return fmt.Sprintf("%s://%s", scheme, cr.Spec.API.Host)
 }
+
+func handleGenericUserCredentialsSecret(username, password, usernameKey, passwordKey, secretName string, cr *apiv1alpha1.Astarte, c client.Client, scheme *runtime.Scheme) error {
+	createUserCredentialsSecret := true
+	createUserCredentialsSecretFromCredentials := false
+	if cr.Spec.Cassandra.Connection != nil {
```

Comment:
Watch out for this. You're referring explicitly to Cassandra rather than checking what was given in the params. Here is where you want all the specs (RMQ and Cassandra) to refer to `LoginCredentialsSecret`, and pass it as an arg here.